Sources/UI/TruncatingLabel/TruncatingLabel.swift

```diff
@@ -0,0 +1,325 @@
+        precondition(primary2 == dots as NSString)
+        primary1 = primaryText[..<firstSplitIndexWithDots].trimmingCharacters(in: .whitespaces) + dots as NSString
+      }
+      /// 1. draw the first line
+      primary1.draw(
+        at: .zero,
+        withAttributes: primaryTextAttributes
+      )
+      /// 2. draw the second line
+      secondaryString.draw(
+        at: .init(
+          x: secondaryStartPosition,
+          y: primarySize.height + offset + (primarySize.height - secondarySize.height)*0.5),
+        withAttributes: secondaryTextAttributes
+      )
+    } else {
+      /// 1. draw the first line
+      primary1.draw(
+        at: .zero,
+        withAttributes: primaryTextAttributes
+      )
+      /// 2. draw the second line
+      /// i)
+      primary2.draw(
+        at: .init(x: 0, y: primarySize.height + offset),
+        withAttributes: primaryTextAttributes
+      )
+      /// ii)
+      secondaryString.draw(
+        at: .init(
+          x: secondaryStartPosition,
+          y: primarySize.height + offset + (primarySize.height - secondarySize.height)*0.5),
+        withAttributes: secondaryTextAttributes
+      )
+    }
+    
+    /// update intrinsic content size
+    internalIntrinsicContentSize = .init(
+      width: frame.width,
+      height: primarySize.height + offset + max(primarySize.height, secondarySize.height)
+    )
+  }
+  
+  public override var intrinsicContentSize: CGSize {
+    internalIntrinsicContentSize
+  }
+}
+
+public extension TruncatingLabel {
+  // @NOTE: - Control the alignment of the secondary text:
+  // When set to `left`, the secondary label will gravitate toward
+  // left (either touching primary text or the left edge):
+  //  -------------------------------------------
+  // |   This is a short text. secondary label   |
+  //  -------------------------------------------
+  //  -------------------------------------------
+  // |   This is a longer single-line text.      |
+  // |   secondary label                         |
+  //  -------------------------------------------
+  //  -------------------------------------------
+  // |   This is a longer two-line text, lorem   |
+  // |   ipsum dolor. secondary label            |
+  //  -------------------------------------------
+  //
+  // When set to `right`, the secondary label will gravitate toward
+  // right:
+  //  -------------------------------------------
+  // |  This is a short text.   secondary label  |
+  //  -------------------------------------------
+  //  -------------------------------------------
+  // |  This is a longer single-line text.       |
+  // |                          secondary label  |
+  //  -------------------------------------------
+  //  -------------------------------------------
+  // |  This is a longer two-line text, lorem    |
+  // |  ipsum dolor.            secondary label  |
+  //  -------------------------------------------
+  enum Gravity {
+    case left, right
```

Comment:
Add a comment for each case as well.
```swift
/// `secondaryText` sticks to the end of the `primaryText`
case left
/// `secondaryText` sticks to the right edge of this view
case right
```

Reply:
Maybe rename `left/right` to `leading/trailing` if we are going to support RTL languages some day 🤷‍♂️. Don't have strong feelings about that though.